### PR TITLE
Add workflow for ACR/MCR image build and publish.

### PR DESCRIPTION
**Pull request by mattstam**
This will set up Github action ACR publish (which eventually gets to MCR). Using @Tatsinnit's [PR](https://github.com/Azure/aks-periscope/pull/151) as example.

Build has been tested in fork: https://github.com/mattstam/ip-masq-agent-v2/runs/5415657376?check_suite_focus=true

Thanks to @Tatsinnit and @peterbom for organizing this flow and helping with set up 🙏

### CODE REVIEW

File: .github/workflows/build-publish-mcr.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        uses: azure/login@v1
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+      - name: 'Run Azure CLI commands'
+        run: |
+          az acr login -n ${{ secrets.AZURE_REGISTRY_SERVER }}
+          REGISTRY=${{ secrets.AZURE_REGISTRY_SERVER }}/public/aks VERSION=${{ steps.changelog_reader.outputs.version }} make push
+          #docker tag public/aks/ip-masq-agent-v2 ${{ secrets.AZURE_REGISTRY_SERVER }}/public/aks/ip-masq-agent-v2:${{ steps.changelog_reader.outputs.version }}
```
Comment (peterbom) [inline .github/workflows/build-publish-mcr.yml:67]: worth removing the commented-out lines?

File: Makefile [hunk 2]
```diff
@@ -64,7 +64,7 @@ ifeq ($(ARCH),ppc64le)
     BASEIMAGE?=k8s.gcr.io/build-image/debian-iptables-ppc64le:buster-v1.7.0
 endif
 
-TAG := $(VERSION)__$(OS)_$(ARCH)
+TAG := $(VERSION)
```
Comment (peterbom) [inline Makefile:67]: any scenarios where removing the OS and ARCH values might cause any issues? An alternative approach might be to use the `?=` syntax to keep the existing behaviour but override it for this particular case?

File: .github/workflows/build-publish-mcr.yml (hunk 1 above)
Comment (mattstam) [reply .github/workflows/build-publish-mcr.yml]: Yup, fixed!

File: Makefile (hunk 2 above)
Comment (mattstam) [reply Makefile:67]: I think it should be okay for now, was mainly just for pushing to personal registry I think, and it's probably unusual to have this in official image tag. Might revisit this later when multi-arch support becomes more of a concern.

**Review by Tatsinnit — APPROVED** [review]
👍 - Thank you! I think we tested this against my test `acr` please don't trigger this until we have the `mcr` pr merged

**Review by peterbom — APPROVED** [review]

**Pull request merged**